R/knitr-hooks.R

```diff
@@ -191,8 +243,24 @@ install_knitr_hooks <- function() {
           preserved_options$exercise.df_print <- "default"
         preserved_options$exercise.timelimit <- options$exercise.timelimit
         preserved_options$exercise.setup <- options$exercise.setup
-        preserved_options$exercise.checker <- deparse(options$exercise.checker)
 
+        root_chunk <- get_knitr_chunk(options$label)
+        root_options <- attr(root_chunk, "chunk_opts")
+        preserved_options$engine <- root_options$engine
```

Comment:
Is this able to be done with 
```suggestion
        preserved_options$engine <- options$engine %||% "r"
```
?

Reply:
If so, can we make a small helper method and use it in the other places we do this?

```r
knitr_engine <- function(engine) {
  engine %||% "r"
}
```

Reply:
Yes, that works actually. Previously I thought I needed more information from the root knitr chunk but since we just need the engine, that will work. I think the helper method would come in handy so I'll make one.